File: task-03/python/sub-task 4.py
num = int(input("Enter a number: "))
file = open('input.txt', 'w')
file.write(str(num))
file.close()

input_file = open("input.txt", 'r')
content = input_file.read()
nnum = int(content)
input_file.close()

output = open("output.txt", 'w')

for i in range(nnum):
    for j in range(nnum - i - 1):
        output.write(" ")
    for j in range(i + 1):
        output.write("* ")
    output.write("\n")

for i in range(nnum - 1):
    for j in range(i + 1):
        output.write(" ")
    for j in range(nnum - i - 1):
        output.write("* ")
    output.write("\n")

output.close()
file = open("output.txt", 'r')
patt = file.read()
file.close()
print(patt)

File: task-03/ruby/Sub-task 4.rb
print "Enter a number: "
num = gets.to_i
File.open('input.txt', 'w') do |file|
  file.write(num.to_s)
end
content = File.read('input.txt')
nnum = content.to_i

File.open('output.txt', 'w') do |output|
  for i in 0...nnum
    for j in 0...(nnum - i - 1)
      output.write(" ")
    end
    for j in 0..i
      output.write("* ")
    end
    output.write("\n")
  end

  for i in 0...(nnum - 1)
    for j in 0..i
      output.write(" ")
    end
    for j in 0...(nnum - i - 1)
      output.write("* ")
    end
    output.write("\n")
  end
end

patt = File.read('output.txt')
puts patt
